pulsar/producer_partition.go

```diff
@@ -79,6 +79,9 @@ type partitionProducer struct {
 	schemaInfo       *SchemaInfo
 	partitionIdx     int32
 	metrics          *internal.TopicMetrics
+
+	epoch                    uint64
+	userProvidedProducerName bool
```

Comment:
Can this `userProvidedProducerName`  just be determined when creating the producer command from the name? Why create an extra variable?